tools/logger/convert.c

```diff
@@ -673,30 +673,11 @@ static int verify_fw_ver(const struct convert_config *config,
 		return -ferror(config->version_fd);
 	}
 
-	ret = memcmp(&ver, &snd->version, sizeof(struct sof_ipc_fw_version));
-	if (ret) {
-		log_err(config->out_fd,
-			"fw version in %s file does not coincide with fw version in %s file.\n",
-			config->ldc_file, config->version_file);
-		return -EINVAL;
-	}
-
-	/* logger and version_file abi dbg verification */
-	if (SOF_ABI_VERSION_INCOMPATIBLE(SOF_ABI_DBG_VERSION,
-					 ver.abi_version)) {
-		log_err(config->out_fd,
-			"abi version in %s file does not coincide with abi version used by logger.\n",
-			config->version_file);
-		log_err(config->out_fd,
-			"logger ABI Version is %d:%d:%d\n",
-			SOF_ABI_VERSION_MAJOR(SOF_ABI_DBG_VERSION),
-			SOF_ABI_VERSION_MINOR(SOF_ABI_DBG_VERSION),
-			SOF_ABI_VERSION_PATCH(SOF_ABI_DBG_VERSION));
+	/* compare firmware hash value from version file and ldc file */
+	if (ver.fw_hash != snd->version.fw_hash) {
```

Comment:
Should the verification be mandatory now, as it is more reliable (-e assumed when reading live trace and -v assumed when reading from a file)?

Reply:
It sounds good for me, maybe then some flag, e.g. -E should be added to force running without compatibility check